aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -124,6 +126,14 @@ public void configure(
         final AtomicReference<Exception> asyncException = new AtomicReference<>();
         LogFactory.setLevel(LogFactory.Level.OFF);
 
+        try {
+            pluginConfiguration.put("UserAgentOverride", UserAgent.string());
```

Comment:
It's a bad practice to mutate method arguments. Here, the following occurs:
1. `pluginConfiguration` has state A.
2. `configure(..., pluginConfiguration, ...)` runs.
3. `pluginConfiguration` has state B.

Instead, I would do:
```java
JSONObject mobileClientConfig = new JSONObject(pluginConfiguration);
mobileClientConfig.put("UserAgentOverride", UserAgent.string());
```

Reply:
Done